lxd/instance/drivers/driver_lxc.go

```diff
@@ -8536,11 +8533,7 @@ func (d *lxc) getFSStats() (*metrics.MetricSet, error) {
 	}
 
 	// Get disk devices
-	for _, dev := range d.expandedDevices {
-		if dev["type"] != "disk" || dev["path"] == "" {
-			continue
-		}
-
+	for _, dev := range d.expandedDevices.Filter(filters.IsDisk, filters.HasPath) {
```

Comment:
So this sort of construct isn't really delivering any value in my mind, because its just replicating the logic we already had, just in a more abstracted form.

What I was hoping for was more of a set of higher *definition of concepts* filters.

So rather than filtering for "device type disk with non-empty path", we could instead ask "why are we asking for devices with those properties - what does it represent to us?'

Using that mindset, and from just a cursory glance, this function is getting metrics for *filesystem disks* (and so it is using a non-empty `path` as a proxy for that meaning).

So how about `d.expandedDevices.Filter(filters.FilesystemDisk)`?

Reply:
So `IsRootDisk` is great. And general device type filters like `IsDisk` and `IsNIC` are useful in that they allow using the filtering method consistently, but be cautious of not re-inventing the logic we already have in multiple places.